doc/mgr/smb.rst

```diff
@@ -0,0 +1,644 @@
+
+Print a listing of cluster ids. The output defaults to JSON, select YAML
+encoding with the ``--format=yaml`` option.
+
+
+Share Commands
+--------------
+
+Create Share
+++++++++++++
+
+.. code:: bash
+
+    $ ceph smb share create <cluster_id> <share_id> <cephfs_volume> <path> [--share-name=<share_name>] [--subvolume=<subvolume>] [--readonly]
+
+Create a new SMB share, hosted by the named cluster, that maps to the given
+CephFS volume and path.
+
+Options:
+
+cluster_id
+    A short string uniquely identifying the cluster
+share_id
+    A short string uniquely identifying the share
+cephfs_volume
+    The name of the cephfs volume to be shared
+path
+    A path relative to the root of the volume and/or subvolume
+share_name
+    Optional. The public name of the share, visible to clients. If not provided
+    the ``share_id`` will be used automatically
+subvolume
+    Optional. A subvolume name in the form ``[<subvolumegroup>/]<subvolume>``.
+    The option causes the path to be relative to the CephFS subvolume
+    specified.
+readonly
+    Creates a read-only share
+
+Remove Share
+++++++++++++
+
+.. code:: bash
+
+    $ ceph smb share rm <cluster_id> <share_id>
+
+Remove an SMB Share from the cluster.
+
+
+List Shares
++++++++++++
+
+.. code:: bash
+
+    $ ceph smb share ls <cluster_id> [--format=<format>]
+
+Print a listing of share ids. The output defaults to JSON, select YAML
+encoding with the ``--format=yaml`` option.
+
+.. _mgr-smb-declarative:
+
+Management Commands - Declarative Style
+=======================================
+
+In addition to the basic imperative management commands the ``smb`` manager
+module supports configuration using declarative resource specifications.
+Resource specifications can be written in either JSON or YAML. These resource
+specifications can be applied to the cluster using the ``ceph smb apply``
+command, for example:
+
+.. code:: bash
+
+    $ ceph smb apply -i /path/to/resources.yaml
+
+Resources that have already been applied to the Ceph cluster configuration can
+be viewed using the ``ceph smb show`` command. For example:
+
+.. code:: bash
+
+    $ ceph smb show [<resource_name>...]
+
+The ``show`` command can show all resources of a given type or specific
+resources by id. ``resource_name`` arguments can take the following forms:
+
+- ``ceph.smb.cluster``: show all cluster resources
+- ``ceph.smb.cluster.<cluster_id>``: show specific cluster with given cluster id
+- ``ceph.smb.share``: show all share resources
+- ``ceph.smb.share.<cluster_id>``: show all share resources part of the given
+  cluster
+- ``ceph.smb.share.<cluster_id>.<share_id>``: show specific share resource with
+  the given cluster and share ids
+- ``ceph.smb.usersgroups``: show all Users & Groups resources
+- ``ceph.smb.usersgroups.<users_goups_id>``: show a specific Users & Groups
+  resource
+- ``ceph.smb.join.auth``: show all join auth resources
+- ``ceph.smb.join.auth.<auth_id>``: show a specific join auth resource
```

Comment:
I considered that, but I felt like the very explicit scoping was a benefit. If, as I hope, that one day other modules adopt similar approaches I think it would be nice for all the "types" to be namespaces. I have, and will reconsider, making the "fully qualified" names optional later. But I think I'd like to get some real-world use with this more explicit version and then support automatically aliasing short forms later if enough people want it.